core/services/sleeper_task_test.go

```diff
@@ -72,18 +94,71 @@ func TestSleeperTask_SenderNotBlockedWhileWorking(t *testing.T) {
 	sleeper.Stop()
 }
 
-func TestSleeperTask_StopWithoutStartNonBlocking(t *testing.T) {
+// TODO: Do we really need to support restarting? It doesnt't appear to be used anywhere
```

Comment:
There is a (failing) test here for restart, but this doesn't actually seem to actually be used anywhere in the app. Can we remove the test?